match2/commons/template_match.lua

```diff
@@ -0,0 +1,112 @@
+local p = {}
```

Comment:
Could you rename `p` to something more descriptive?

Reply:
i hove `TemplateMatch` instead of `p` is good^^